Frontend/nutri-pal/src/app/diary/page.tsx

```diff
@@ -0,0 +1,223 @@
+import '@/styles/diary.css'
+import { useEffect, useState } from "react"
+import axios from "axios"
+import { onAuthStateChanged } from "firebase/auth"
+import { auth } from "../firebase-config"
+import { BASE_URL_REST_API } from "../consts"
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import { FormControl, InputLabel } from "@mui/material"
+
+export default function Diary() {
+    const [user, setUser] = useState<any>({})
+    const [getFlag, setGetFlag] = useState(false)
+    const [dates, setDates] = useState<any>(null)
+
+    const [openModal, setOpenModal] = useState(false)
+    const [modalMeal, setModalMeal] = useState<any>(null);
+
+    const [maxDate, setMaxDate] = useState('');
+    const [minDate, setMinDate] = useState('');
+    const [rangeYear, setRangeYear] = useState<any>([]);
+
+    const [filterDate, setFilterDate] = useState('');
+    const [filterYear, setFilterYear] = useState('');
+    const [filterMonth, setFilterMonth] = useState('');
+    const [monthFlag, setMonthFlag] = useState(false);
+
+    // Dictionary to get months by its number
+    const months_dictionary: { [key: number]: string } = {
+        1: "Jan",
+        2: "Feb",
+        3: "Mar",
+        4: "Apr",
+        5: "May",
+        6: "June",
+        7: "July",
+        8: "Aug",
+        9: "Sept",
+        10: "Oct",
+        11: "Nov",
+        12: "Dec"
+    };
+
+    // Array to display the months on the Select for filtering
+    const monthsOptions = [
+        { value: '', label: 'Filter by Month' },
+        { value: '01', label: 'January' },
+        { value: '02', label: 'February' },
+        { value: '03', label: 'March' },
+        { value: '04', label: 'April' },
+        { value: '05', label: 'May' },
+        { value: '06', label: 'June' },
+        { value: '07', label: 'July' },
+        { value: '08', label: 'August' },
+        { value: '09', label: 'September' },
+        { value: '10', label: 'October' },
+        { value: '11', label: 'November' },
+        { value: '12', label: 'December' }
+    ]
+
+    // Function tu turn the ISO date into "January 01, 2024" style
+    const isoToDate = (iso: string) => {
+        const dateArray = iso.split('-')
+
+        return (`${months_dictionary[parseInt(dateArray[1])]} ${dateArray[2].slice(0, 2)}, ${dateArray[0]}`)
+    }
+
+    // Setting the user that was authenticated by Firebase to a variable and declare the flag to start making request to true
+    onAuthStateChanged(auth, (currentUser) => {
+        if (currentUser) {
+            setUser(currentUser);
+            setGetFlag(true)
+        }
+    })
+
+    // Do a GET request for the registered days that the user has with or without the query params
+    const getDates = (year = '', month = '', day = '') => {
+        const HISTORY_DAYS = BASE_URL_REST_API + 'history/registered_days/' + user.uid + `?year=${year}&month=${month}&day=${day}`;
```

Comment:
it might make sense to consolidate all api creation urls into a helper function so we don't need to dig through all of our files to know/update routes